vhdl_lang/src/analysis/completion.rs

```diff
@@ -63,9 +63,13 @@ struct PortsOrGenericsExtractor {
 
 impl DesignRoot {
     fn extract_port_or_generic_names(&self, id: EntityId, kind: MapAspectKind) -> Vec<EntityId> {
-        let mut searcher = PortsOrGenericsExtractor::new(id, kind);
-        self.walk(&mut searcher);
-        searcher.items
+        if let Some(ref pos) = self.get_ent(id).decl_pos {
+            let mut searcher = PortsOrGenericsExtractor::new(id, kind);
```

Comment:
Now that I look closer at what the PortsOrGenericsExtractor does it could be a lot simpler. You do not need to go via the AST to find the ports or generics of an entity. 

The `AnyEntKind::Component` contains a `Region` which contains all the `InterfaceEnt` for ports generics.
https://github.com/VHDL-LS/rust_hdl/blob/master/vhdl_lang/src/analysis/named_entity.rs#L45

The `AnyEntKind::Design(Design::Entity)` contains a `Region` which is all the ports and generics:
https://github.com/VHDL-LS/rust_hdl/blob/master/vhdl_lang/src/analysis/named_entity/design.rs#L23